lib/internal/process/next_tick.js

```diff
@@ -1,47 +1,9 @@
 'use strict';
 
-// This value is used to prevent the nextTickQueue from becoming too
-// large and cause the process to run out of memory. When this value
-// is reached the nextTimeQueue array will be shortened (see tickDone
-// for details).
-const kMaxCallbacksPerLoop = 1e4;
```

Comment:
Doesn't this expose infinite nextTick loop problems? and / or, isn't this a breaking change?

Reply:
This became irrelevant when @mscdex turned this into a linked list instead of an Array. I don't really see that it does anything right now other than make sure the indexes don't overflow and since I got rid of length & index, it's no longer necessary.

Reply:
Also, FWIW, our benchmarks are an indirect test for this and they still pass. (If one were to remove this without removing the index & length values then the integers overflow.)

Reply:
I thought this was to prevent infinite `nextTick` recursion... I'm not sure how a linkedlist helps that? Maybe I didn't look deep enough? Not sure.

Reply:
No, it doesn't prevent infinite recursion. Nothing does, our documentation even calls it out as an issue.

When an Array used to be used for `nextTick`, we wouldn't clear it until 1e4 runs so it could grow to near Infinite size without this condition to remove stale tick objects.

Reply:
> Note: The next tick queue is completely drained on each pass of the event loop before additional I/O is processed. As a result, recursively setting nextTick callbacks will block any I/O from happening, just like a while(true); loop.

https://nodejs.org/dist/latest-v9.x/docs/api/process.html#process_process_nexttick_callback_args

Reply:
@Fishrock123 here's the PR that introduced this code https://github.com/apapirovski/node/commit/5757642e912c0a4de6476fc766748b6e33ac845b

Reply:
@apapirovski That was a very helpful explanation… on some level it’s sad to understand code only once it’s going away, but I always wondered about this. :)